File: grabscrab/grabscrab.js
const grabscrab = (string, dict) => {
    const alphabetise = (str) => {
        return str.split('').sort((a, b) => {
            return a.localeCompare(b);
        });
    };

    const sort = alphabetise(string);
    const output = [];
    dict.forEach((element) => {
        const check = alphabetise(element);
        if(check.join('') === sort.join('')){
            output.push(element);
        };
    });

    return output;
};

module.exports = { grabscrab };

//https://www.codewars.com/kata/52b305bec65ea40fe90007a7/train/javascript

File: python-venv/grabscrab/kata_function/grabscrab.py
def grabscrab(string, array):
    sort = sorted(list(string))
    output = []

    for i in range(0, len(array), 1):
        check = sorted(list(array[i]))
        if ''.join(sort) == ''.join(check):
            output.append(array[i])

    return output


#https://www.codewars.com/kata/52b305bec65ea40fe90007a7/train/python